src/tests/profiler/native/README.md

```diff
@@ -33,18 +33,44 @@ When you want to test new profiler APIs you will need a new test profiler implem
 
 1) Get your new profiler building:
 
- - Copy and rename gcbasicprofiler folder.
+ - Copy and rename gcbasicprofiler folder for the native part.
  - Rename the source files and the gcbasicprofiler type within the source.
  - Add the new source files to CMakeLists.txt
+ - Copy and rename gc managed folder for the managed part.
+ - Rename the gc.cs and gc.csproj and update the profiler GUID + test name.
+   static readonly Guid YourProfilerGuid = new Guid("XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX");
+   ...
+       return ProfilerTestRunner.Run(profileePath: System.Reflection.Assembly.GetExecutingAssembly().Location,
+           testName: "YourFeature",
+           profilerClsid: YourProfilerGuid);
 
 2) Make your new profiler creatable via COM:
 
  - Create a new GUID and replace the one in YourProfiler::GetClsid()
- - Update classfactory.cpp to include your new profiler's header and update the list of profiler instances in ClassFactory::CreateInstance
-
-        Profiler* profilers[] = {
-    		new GCBasicProfiler(),
-    		// add new profilers here
-    	};
-
-3) Override the profiler callback functions that are relevant for your test and delete the rest. At minimum you will need to ensure that the test prints the phrase "PROFILER TEST PASSES" at some point to indicate this is a passing test. Typically that occurs in the Shutdown() method. It is also likely you want to override Initialize() in order to call SetEventMask so that the profiler receives events.
+ - Update classfactory.cpp to include your new profiler's header and add its GUID to the list of profiler instances in ClassFactory::CreateInstance
```

Comment:
Thanks for updating the docs!

Reply:
> Thanks for updating the docs!

I'm starting to look at the MS Docs too  ;^)